obspy/core/tests/test_stream.py

```diff
@@ -2656,6 +2656,59 @@ def test_write_empty_stream(self):
             self.assertEqual(e.exception.args[0],
                              'Can not write empty stream to file.')
 
+    def test_stack(self):
```

Comment:
There is a lot going on here, do you mind breaking this up into a few different tests? Or maybe just add a few more  comments about what each subtest is doing if you prefer.